sharing_portal/views.py

```diff
@@ -325,23 +378,24 @@ def artifact(request, artifact, artifact_versions, version_idx=None):
         }
 
     # Ensure launch URLs are authenticated if a private link is being used.
-    if SHARING_KEY_PARAM in request.GET:
-        launch_url += '?{}={}'.format(SHARING_KEY_PARAM, request.GET[SHARING_KEY_PARAM])
+    request_day_pass_url = preserve_sharing_key(
+        reverse("sharing_portal:request_day_pass", args=[artifact.pk]), request
+    )
 
-    template = loader.get_template('sharing_portal/detail.html')
+    template = loader.get_template("sharing_portal/detail.html")
 
     context = {
-        'artifact': artifact,
-        'all_versions': _artifact_display_versions(artifact_versions),
-        'doi_info': doi_info,
-        'version': version,
-        'launch_url': launch_url,
-        'related_artifacts': artifact.related_items,
-        'editable': (
-            request.user.is_staff or (
-            artifact.created_by == request.user)),
+        "artifact": artifact,
+        "all_versions": _artifact_display_versions(artifact_versions),
+        "doi_info": doi_info,
+        "version": version,
+        "launch_url": launch_url,
```

Comment:
🚨  Make sure the sharing key is included in the launch URL as well

Reply:
Fixed

Reply:
Thanks!